### PR TITLE
(CM-51) Create a presenter for contact blocks

### PR DESCRIPTION
**Pull request by pezholio**
This creates a presenter for contact blocks - if no field is provided, we show all the values (in a similar way to how Whitehall present contacts), otherwise we present a field. I've also added the concept of a field presenter, which allows us to present email addresses as linked.

There's also some improvements to testing, so we don't get caught out by whitespace in resulting HTML

### CODE REVIEW

File: spec/content_block_tools/content_block_spec.rb [hunk 1]
```diff
@@ -50,6 +50,16 @@
       end
     end
 
+    context "when a contact" do
+      let(:document_type) { "content_block_contact" }
+      let(:presenter_class) { ContentBlockTools::Presenters::ContactPresenter }
+
+      it "calls the email address presenter" do
```
Comment (Harriethw) [inline spec/content_block_tools/content_block_spec.rb:57]: minor typo: should this be "calls the contact presenter?"

File: lib/content_block_tools/presenters/base_presenter.rb [hunk 2]
```diff
@@ -26,7 +26,7 @@ def initialize(content_block)
       # @return [string] A HTML representation of the content block
       def render
         content_tag(
-          self.class::BASE_TAG_TYPE,
+          field_names ? :span : self.class::BASE_TAG_TYPE,
```
Comment (Harriethw) [inline lib/content_block_tools/presenters/base_presenter.rb:29]: minor: could see this stray `:span` logic getting a bit lost - could it be a method in itself, and maybe the `:span` is a constant on the field presenter?

File: lib/content_block_tools/presenters/base_presenter.rb [hunk 3]
```diff
@@ -78,6 +78,10 @@ def field_names
           all_fields&.split("/")&.map(&:to_sym)
         end
       end
+
+      def field_presenter
+        @field_presenter ||= field_names ? self.class::FIELD_PRESENTERS[field_names.last] : nil
```
Comment (Harriethw) [inline lib/content_block_tools/presenters/base_presenter.rb:83]: minor: i'm realising we're using `field_names` a lot now, I know you haven't really touched the code here but i wonder if it should or could be memoized

**Review by Harriethw — APPROVED** [review]
awesome! great improvements to the tests. just a few nit picking comments really
